### PR TITLE
Add tests for workers

### PR DESCRIPTION
**Pull request by Shinigami92**
## PR Checklist

- [x] Please add a description of your changes.
- [x] We need your changes to come with unit tests in order to keep this project in quality and easy to maintain.
- [ ] Please add a link to the open issue or task that this pull request will resolve.
- [ ] Add a note to the changelog to indicate the change.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

If yes please indicate it and migration info into the CHANGELOG.md file.

---

I add some tests for thread-worker and cluster-worker

Sadly the `ClusterWorker` cannot be tested that easy, because it is in main mode when just created

### CODE REVIEW

**Comment by Shinigami92** [comment]
Goal of this PR is to increase the coverage of thread-worker

image

File: tests/pools/cluster/dynamic.test.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const max = 3
 const pool = new DynamicClusterPool(
   min,
   max,
-  './tests/worker/cluster/testWorker.js',
+  './tests/worker-files/cluster/testWorker.js',
```
Comment (jerome-benoit) [inline tests/pools/cluster/dynamic.test.js:8]: It's more the worker scripts here.

Comment (Shinigami92) [reply tests/pools/cluster/dynamic.test.js:8]: I also thought about the name `fixtures`
I think this is a common name for files for tests that are used within them, but not the tests directly

File: tests/worker/thread-worker.test.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const expect = require('expect')
```
Comment (pioardi) [inline tests/worker/thread-worker.test.js:1]: Thanks @Shinigami92  was really important for me to add a test for the thread worker.
I see also that you removed some code duplications thank you thank you, thread worker was covered but unfortunately nyc does not recognize code covered during other threads run.

Anyway we could add a test for the sendToMainWorker, we will need to mock something I guess

**Review by pioardi — COMMENTED** [review]
Just a comment to add another small test and we have done

**Comment by Shinigami92** [comment]
> Just a comment to add another small test and we have done

I think I will look into that tomorrow :slightly_smiling_face:  
Need to read about how to mock this.

**Comment by pioardi** [comment]
Sounds good, we are now at 90% few other test cases and we have done bro, I was also preparing something but you are too fast for me lol

**Comment by Shinigami92** [comment]
@pioardi 
Basically this branch just enhance the tests folder structure
IMO you can squash-merge it so we have this in the master

File: tests/worker/thread-worker.test.js (hunk 2 above)
Comment (pioardi) [reply tests/worker/thread-worker.test.js:1]: Do you want to do this before to merge or do you prefer to handle this later?

Comment (Shinigami92) [reply tests/worker/thread-worker.test.js:1]: IMO we should merge it and handle this later

Comment (pioardi) [reply tests/worker/thread-worker.test.js:1]: k

**Review by pioardi — APPROVED** [review]
Approved

**Pull request merged**